ethcore/src/client/client.rs

```diff
@@ -556,7 +564,7 @@ impl Client {
 			BlockID::Number(number) => Some(number),
 			BlockID::Hash(ref hash) => self.chain.block_number(hash),
 			BlockID::Earliest => Some(0),
-			BlockID::Latest => Some(self.chain.best_block_number())
+			BlockID::Latest | BlockID::Pending => Some(self.chain.best_block_number()),
```

Comment:
client has more immediate access to miner (through an `Arc`) and already manages other splits.